src/CMakeLists.txt

```diff
@@ -72,31 +72,47 @@ if (CMAKE_BUILD_TYPE MATCHES Debug)
 endif (CMAKE_BUILD_TYPE MATCHES Debug)
 
 include(build-defs.cmake)
-set(GITCOMMIT_H ${CMAKE_CURRENT_BINARY_DIR}/gitcommit.h)
 
-if (WIN32)
-  add_custom_command(
-    OUTPUT ${GITCOMMIT_H}
-    WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
-    COMMAND ${CMAKE_COMMAND} -E echo_append "#define EXT_GIT_COMMIT " > ${GITCOMMIT_H}
-    COMMAND (${GIT_EXECUTABLE} describe --abbrev=4 --dirty --always --tags 2> NUL || call && echo "${PROJECT_VERSION_MOD}") >> ${GITCOMMIT_H}
-    COMMENT "Generating gitcommit.h"
-    VERBATIM)
-else ()
-  add_custom_command(
-    OUTPUT ${GITCOMMIT_H}
-    WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
-    COMMAND ${CMAKE_COMMAND} -E echo_append "#define EXT_GIT_COMMIT " > ${GITCOMMIT_H}
-    COMMAND sh -c "if `${GIT_EXECUTABLE} status > /dev/null 2>&1`; then ${GIT_EXECUTABLE} describe --abbrev=4 --dirty --always --tags ; else echo ${PROJECT_VERSION_MOD} ; fi" >> ${GITCOMMIT_H}
-    COMMENT "Generating gitcommit.h"
-    VERBATIM)
-endif (WIN32)
+if(GIT_FOUND)
```

Comment:
There were an explanation in the PR, albeit terse. As we discussed, the code is now updated to return an explicit error instead of just erroring with a "function not defined", so I updated the PR comment and the commit description to give a more elaborate description.